File: retrieve_data/download_data.R
# TODO: Set path to receive downloads on your local machine
working_directory <- "YOUR-WORKING-DIRECTORY-HERE"
setwd(working_directory)

# Load necessary libraries
library(aws.s3)
library(jsonlite)

# TODO: Set environment variables for bucket access
Sys.setenv(
  "AWS_ACCESS_KEY_ID" = "",
  "AWS_SECRET_ACCESS_KEY" = "",
  "AWS_DEFAULT_REGION" = "us-east-1"
)

# Set additional variables for use
bucket.name <- "sportradar-wisd-data"

# Create a dataframe with filenames and other relevant information
files.df <- get_bucket_df(bucket.name)

# Create a vector of filenames to be downloaded
filenames <- files.df$Key

# Iterate through the files and download them to your local working directory
for(filename in filenames){
  save_object(object = filename,
              bucket = bucket.name)
}

# To load a single downloaded file into R memory, you can use the following line
# To load all the downloaded files into R memory, you can loop through them again
sample.df <- stream_in(file(basename(filename)))

File: retrieve_data/download_data.py
import boto3


def main():
    """
    Simple script to download wisd sportsradar data to local directory
    """
    s3_client = boto3.client('s3')
    bucket = "sportradar-wisd-data"
    prefix = "games"

    print("Connecting to Sportradar S3 bucket...")
    response = s3_client.list_objects_v2(Bucket=bucket, Prefix=prefix)
    files = response.get("Contents")
    print(f"Success! \nDownloading {len(files)} files... \n")
    for file_num, file in enumerate(files):
        current_file = file['Key']
        split_path = current_file.split("/")
        file_to_copy = split_path[2]
        print(
            f"Downloading file: {file_to_copy} [{file_num + 1}/{len(files)}]"
        )
        s3_client.download_file(bucket, current_file, file_to_copy)
        print(f"Successfully downloaded {file_to_copy} \n")


if __name__ == "__main__":
    main()
